### PR TITLE
PLANNER-945: Renamed some REST paths to remove JAX-RS warnings

### PR DESCRIPTION
**Pull request by Christopher-Chianelli**

### CODE REVIEW

**Comment by tiagobento** [comment]
Which warnings do these changes remove? It seems not ideal to have a `/get` at the end of our endpoints..

**Comment by Christopher-Chianelli** [comment]
@tiagobento Ambiguous method names warning (getSpot REST API name conflicts with getSpotGroups REST API name, getShift REST API name conflicts with getTemplate REST API NAME)

**Comment by tiagobento** [comment]
@Christopher-Chianelli Ok, now I understand the issue. It seems to me that a better solution would be adding RegExps to the methods that take integers as parameters. This way, they won't clash with the strings that are part of the other endpoints. E.g. `@Path("{id : \\d+}")`, where `\\d` is a shortcut for `[0-9]`. What do you think?

**Comment by Christopher-Chianelli** [comment]
@tiagobento I think I tried that already, it either failed to work or still got the warning (need to retest), but if it works, then I think that is a better solution.

**Comment by ge0ffrey** [comment]
I agree with the above - the rest path shouldn't include get.
But do note if we would create for example today:
/employee/<country>
that we can't later create (without breaking backwards compatibility at least)
/employee/<age>
so we should think ahead, by creating something like
/employee/country/<country>
so we can later make something like
/employee/age/<age>

I hope that makes sense.

**Comment by ge0ffrey** [comment]
@Christopher-Chianelli can you show @tiagobento what you tried related to the regex approach and meet up on irc on bluejeans to discuss why it works or doesn't?

**Comment by Christopher-Chianelli** [comment]
@ge0ffrey @tiagobento Retested the regex approach; it works (so either bad memory; version issue, or badly formatted regex for the reason) (I don't have access to my previous attempt unfortunately).

**Comment by tiagobento** [comment]
@ge0ffrey @Christopher-Chianelli Great news :)

**Comment by Christopher-Chianelli** [comment]
@tiagobento redact that tempoary, I realized I only tested with EmployeeRestService, whose methods aren't called; going to test with SpotRestService...

**Comment by Christopher-Chianelli** [comment]
@tiagobento Redacted; here the error message:
Error calling REST method with status (500: Internal Server Error) and text (RESTEASY003210: Could not find resource for full path: http://localhost:8080/rest/tenant/1/spot/groups/getorg.jboss.resteasy.core.registry.SegmentNode.match(SegmentNode.java:119)
org.jboss.resteasy.core.registry.RootNode.match(RootNode.java:43)
org.jboss.resteasy.core.registry.RootClassNode.match(RootClassNode.java:48)
org.jboss.resteasy.core.ResourceMethodRegistry.getResourceInvoker(ResourceMethodRegistry.java:445)
org.jboss.resteasy.core.SynchronousDispatcher.getInvoker(SynchronousDispatcher.java:264)
org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:201)
org.jboss.resteasy.plugins.server.servlet.ServletContainerDispatcher.service(ServletContainerDispatcher.java:221)
org.jboss.resteasy.plugins.server.servlet.HttpServletDispatcher.service(HttpServletDispatcher.java:56)
org.jboss.resteasy.plugins.server.servlet.HttpServletDispatcher.service(HttpServletDispatcher.java:51)
javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
io.undertow.servlet.handlers.ServletHandler.handleRequest(ServletHandler.java:85)
io.undertow.servlet.handlers.security.ServletSecurityRoleHandler.handleRequest(ServletSecurityRoleHandler.java:62)
io.undertow.servlet.handlers.ServletDispatchingHandler.handleRequest(ServletDispatchingHandler.java:36)
org.wildfly.extension.undertow.security.SecurityContextAssociationHandler.handleRequest(SecurityContextAssociationHandler.java:78)
io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
io.undertow.servlet.handlers.security.SSLInformationAssociationHandler.handleRequest(SSLInformationAssociationHandler.java:131)
io.undertow.servlet.handlers.security.ServletAuthenticationCallHandler.handleRequest(ServletAuthenticationCallHandler.java:57)
io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
io.undertow.security.handlers.AbstractConfidentialityHandler.handleRequest(AbstractConfidentialityHandler.java:46)
io.undertow.servlet.handlers.security.ServletConfidentialityConstraintHandler.handleRequest(ServletConfidentialityConstraintHandler.java:64)
io.undertow.security.handlers.AuthenticationMechanismsHandler.handleRequest(AuthenticationMechanismsHandler.java:60)
io.undertow.servlet.handlers.security.CachedAuthenticatedSessionHandler.handleRequest(CachedAuthenticatedSessionHandler.java:77)
io.undertow.security.handlers.NotificationReceiverHandler.handleRequest(NotificationReceiverHandler.java:50)
io.undertow.security.handlers.AbstractSecurityContextAssociationHandler.handleRequest(AbstractSecurityContextAssociationHandler.java:43)
io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
org.wildfly.extension.undertow.security.jacc.JACCContextIdHandler.handleRequest(JACCContextIdHandler.java:61)
io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
io.undertow.servlet.handlers.ServletInitialHandler.handleFirstRequest(ServletInitialHandler.java:292)
io.undertow.servlet.handlers.ServletInitialHandler.access$100(ServletInitialHandler.java:81)
io.undertow.servlet.handlers.ServletInitialHandler$2.call(ServletInitialHandler.java:138)
io.undertow.servlet.handlers.ServletInitialHandler$2.call(ServletInitialHandler.java:135)
io.undertow.servlet.core.ServletRequestContextThreadSetupAction$1.call(ServletRequestContextThreadSetupAction.java:48)
io.undertow.servlet.core.ContextClassLoaderSetupAction$1.call(ContextClassLoaderSetupAction.java:43)
io.undertow.servlet.api.LegacyThreadSetupActionWrapper$1.call(LegacyThreadSetupActionWrapper.java:44)
io.undertow.servlet.api.LegacyThreadSetupActionWrapper$1.call(LegacyThreadSetupActionWrapper.java:44)
io.undertow.servlet.api.LegacyThreadSetupActionWrapper$1.call(LegacyThreadSetupActionWrapper.java:44)
io.undertow.servlet.api.LegacyThreadSetupActionWrapper$1.call(LegacyThreadSetupActionWrapper.java:44)
io.undertow.servlet.api.LegacyThreadSetupActionWrapper$1.call(LegacyThreadSetupActionWrapper.java:44)
io.undertow.servlet.api.LegacyThreadSetupActionWrapper$1.call(LegacyThreadSetupActionWrapper.java:44)
io.undertow.servlet.handlers.ServletInitialHandler.dispatchRequest(ServletInitialHandler.java:272)
io.undertow.servlet.handlers.ServletInitialHandler.access$000(ServletInitialHandler.java:81)
io.undertow.servlet.handlers.ServletInitialHandler$1.handleRequest(ServletInitialHandler.java:104)
io.undertow.server.Connectors.executeRootHandler(Connectors.java:202)
io.undertow.server.HttpServerExchange$1.run(HttpServerExchange.java:805)
java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
java.lang.Thread.run(Thread.java:748)
).
Method that caused the error
```
@GET
@Path("/groups")
List<SpotGroup> getSpotGroups(@ApiParam(required = true) @PathParam("tenantId") Integer tenantId);
```
TBH, the error doesn't make much sense.

**Comment by Christopher-Chianelli** [comment]
@ge0ffrey @tiagobento Look like we need to `mvn clean` + delete local wildfly instance whenever an API changed happened. For some reason GWT kept using the old "get" URL. Maybe it is an issue with the change detector for the GWT maven plugin. Anyway, currently regex for all URL's.

**Comment by tiagobento** [comment]
@Christopher-Chianelli Is it ready to merge?

**Comment by Christopher-Chianelli** [comment]
@tiagobento Mergable, but other classes need to have their paths updated too

**Comment by tiagobento** [comment]
@Christopher-Chianelli Ok, I think we should wait until these classes are updated

**Comment by Christopher-Chianelli** [comment]
@tiagobento I made the changes to the other classes (and fixed a bug with update). PR is ready for merging now.

**Comment by tiagobento** [comment]
@Christopher-Chianelli Great, thanks!

File: optashift-employee-rostering-shared/src/main/java/org/optaplanner/openshift/employeerostering/shared/employee/EmployeeRestService.java [hunk 1]
```diff
@@ -55,27 +53,27 @@ Employee getEmployee(@ApiParam(required = true) @PathParam("tenantId") Integer t
     @POST
     @Path("/add")
```
Comment (mbarkley) [inline optashift-employee-rostering-shared/src/main/java/org/optaplanner/openshift/employeerostering/shared/employee/EmployeeRestService.java]: This wasn't changed in this PR, but I'm wondering isn't it more idiomatic to make this a POST against "/tenant/{tenantId}/employee/"?

File: optashift-employee-rostering-shared/src/main/java/org/optaplanner/openshift/employeerostering/shared/employee/EmployeeRestService.java [hunk 2]
```diff
@@ -55,27 +53,27 @@ Employee getEmployee(@ApiParam(required = true) @PathParam("tenantId") Integer t
...
-            @ApiParam(value = "with no id", required = true) Employee employee);
-    
+                         @ApiParam(value = "with no id", required = true) Employee employee);
+
     /**
      * @param employee never null
      * @return never null, with an updated {@link AbstractPersistable#getVersion()}
      */
     @ApiOperation("Update an employee")
     @POST
```
Comment (mbarkley) [inline optashift-employee-rostering-shared/src/main/java/org/optaplanner/openshift/employeerostering/shared/employee/EmployeeRestService.java]: Isn't it more idiomatic to make this a PUT request against the root employee path (/tenant/{id}/employee/)?

File: optashift-employee-rostering-shared/src/main/java/org/optaplanner/openshift/employeerostering/shared/employee/EmployeeRestService.java [hunk 3]
```diff
@@ -88,58 +86,59 @@ Boolean removeEmployee(@ApiParam(required = true) @PathParam("tenantId") Integer
...
     @Path("/availability/add")
```
Comment (mbarkley) [inline optashift-employee-rostering-shared/src/main/java/org/optaplanner/openshift/employeerostering/shared/employee/EmployeeRestService.java]: Similarly, I think the idiomatic thing for this method would be a POST against "/tenant/{tenantId}/employee/availability/".

**Pull request merged**